### ISSUE TITLE
its_container block os detection

### ISSUE THREAD

**Issue by nanliu** [zuazo/dockerspec#2]
So if I attempt to use dockerspec using docker compose with serverspec os detection:

```ruby
describe docker_compose(compose_yml) do
  its_container(:mysql) do
    if os[:family] == 'redhat'
       describe ...
    end
  end
end
```

This results in the following error `gems/specinfra-2.64.0/lib/specinfra/helper/os.rb:30:in 'detect_os': Specinfra failed os detection. (NotImplementedError)`. Digging a bit deeper, this is because when specinfra attempts to obtain uname via `Specinfra.backend.run_command('uname -m')` the docker_compose backend is missing information on which container to execute the command.

Looking at the docker_compose backend, the container settings comes from `get_config(:docker_compose_container).to_s`, the only problem is I'm not sure how to restore this context in ` rspec/resources/its_container.rb` so the Specinfra.backend.run_command method runs successfully. Any hints or suggestions on how to fix this?

p.s.

If I set the `:docker_compose_container`, `os[:family]` works:

```ruby
set :docker_compose_container, :mysql

os[:family]
```
but I'm hoping to get it working in the `its_container` block so this works with any container.

**Comment by zuazo** [comment]
@nanliu sorry for my delayed answer and thank you for all your information. I'm going to try to search for a way to implement it correctly. But, as you have already seen, it is not an easy task.

**Comment by nanliu** [comment]
@zuazo, thanks for looking into this. I will take another crack at this issue as well.

I recently presented a talk for [Portland Docker user group](https://www.meetup.com/Docker-Portland-OR/events/236739986/) regarding how to use [dockerspec in a container](http://www.slideshare.net/NanLiu1/trust-but-verify-testing-with-docker-containers).

This project received lots positive feedback, and people were really interested in trying it out. I found the source code of the project to be very well documented, but most people attending wasn't aware of Ruby API docs, so maybe worth including a link in the README to: http://www.rubydoc.info/github/zuazo/dockerspec/Dockerspec

**Comment by zuazo** [comment]
@nanliu thank you, man :beer: I don't know what to say. I'm glad to know that this little hack can be useful to others :smiley: I can add a link to your presentation in the README if you want.

As for the documentation, the first badge in the README points to the documentation of the released gem:

[![Documentation](http://img.shields.io/badge/docs-rdoc.info-blue.svg?style=flat)](http://www.rubydoc.info/gems/dockerspec)

Your link (http://www.rubydoc.info/github/zuazo/dockerspec/Dockerspec) is for the master branch documentation.

But if you think the link to the documentation might be clearer in another way, or should we write it down elsewhere, feel free to send a PR. I will accept it with open arms. Maybe we should link both the released and master documentation at the beginning of README?

FYI: Whatever progress I make on the *OS detection* subject, I will write it here.

**Comment by zuazo** [comment]
Hi @nanliu:

Sorry for the delay in my answer. Following your comments and recommendations:

- I fixed OS detection in 8046c8a787a5a8a6a0185be41c107478c41b4472 and d76e2503946bd4f828e8ad911c9679db066a0be7.
- I have added both [released and unreleased documentation to the README](https://github.com/zuazo/dockerspec#documentation).
- I also have added [a link to your presentation](https://github.com/zuazo/dockerspec#presentations).

Thanks for all your help!
